servicex/minio_adapter.py

```diff
@@ -114,31 +114,29 @@ async def download_file(
             )
         )
 
-        async with _semaphore:
-            async with self.minio.resource("s3", endpoint_url=self.endpoint_host) as s3:
-                obj = await s3.Object(self.bucket, object_name)
-                remotesize = await obj.content_length
-                if path.exists():
-                    # if file size is the same, let's not download anything
-                    # maybe move to a better verification mechanism with e-tags in the future
-                    localsize = path.stat().st_size
-                    if localsize == remotesize:
-                        return path.resolve()
-                await obj.download_file(path.as_posix())
+        async with self.minio.resource("s3", endpoint_url=self.endpoint_host) as s3:
+            obj = await s3.Object(self.bucket, object_name)
+            remotesize = await obj.content_length
+            if path.exists():
+                # if file size is the same, let's not download anything
+                # maybe move to a better verification mechanism with e-tags in the future
                 localsize = path.stat().st_size
-                if localsize != remotesize:
-                    raise RuntimeError(f"Download of {object_name} failed")
+                if localsize == remotesize:
+                    return path.resolve()
+            await obj.download_file(path.as_posix(), Config=_transferconfig)
+            localsize = path.stat().st_size
+            if localsize != remotesize:
+                raise RuntimeError(f"Download of {object_name} failed")
         return path.resolve()
 
     @retry(
         stop=stop_after_attempt(3), wait=wait_random_exponential(max=60), reraise=True
     )
     async def get_signed_url(self, object_name: str) -> str:
-        async with _semaphore:
-            async with self.minio.client("s3", endpoint_url=self.endpoint_host) as s3:
-                return await s3.generate_presigned_url(
-                    "get_object", Params={"Bucket": self.bucket, "Key": object_name}
-                )
+        async with self.minio.client("s3", endpoint_url=self.endpoint_host) as s3:
```

Comment:
Will this do ok even when we have a bucket with 50K files in it?

Reply:
This will get throttled on the server side, as before. We can discuss putting in a separate semaphore for that, but the main motivation for this PR is to avoid problems overwhelming the client.

(boto3 doesn't provide a built-in mechanism for limiting the signed URL rates, which is a bit of an indication that Amazon doesn't really see this as a problem for their services)

Reply:
On to testing, then!